apps/flame-defi/app/features/EvmWallet/services/SwapServices/SwapService.ts

```diff
@@ -110,48 +110,6 @@ export class SwapRouter {
     return JSBI.divide(JSBI.multiply(amount, adjustedBasisPoints), basisPoints);
   }
 
-  /**
-   * Checks the token allowance and, if needed, sends an approval transaction.
-   *
-   * @param token - The ERC20 token to approve.
-   * @param amount - The required amount (as a JSBI).
-   * @param walletClient - The connected wallet client (from wagmi/viem).
-   * @param publicClient - The public client for reading blockchain data.
-   */
-  private async approveTokenIfNeeded(
-    token: Token,
-    amount: JSBI,
-    walletClient: WalletClient,
-    publicClient: PublicClient,
-  ): Promise<void> {
-    const signerAddress = walletClient?.account?.address as `0x${string}`;
-    // Use the public client to read the allowance.
-    const currentAllowance = (await publicClient.readContract({
-      address: token.address as `0x${string}`,
-      abi: this.erc20Abi,
```

Comment:
When you moved this to another area in the codebase, it no longer used `this.erc20Abi`, so you need to remove that reference and the abi json if we are no longer using it in favor of viem's abi.